delivery_roulier/models/stock_move_line.py

```diff
@@ -39,3 +39,18 @@ def get_unit_price_for_customs(self):
     def get_sale_order_line(self):
         self.ensure_one()
         return self.move_id.sale_line_id
+
+    def write(self, vals):
+        # OVERRIDE to set `quant_id` if `result_package_id` is set
+        # during `stock.picking`::`_put_in_pack` call
+        if "result_package_id" in vals:
+            for move_line in self.filtered(lambda s: not s.result_package_id):
+                quants_by_product = self.env[
+                    "stock.quant"
+                ]._get_quants_by_products_locations(
+                    move_line.product_id, move_line.location_id
+                )
+                if not quants_by_product or len(quants_by_product) > 1:
+                    continue
+                vals["quant_id"] = quants_by_product.get(move_line.product_id.id).id
```

Comment:
I don't understand here : we are in a for loop, and assigning a value to `vals["quant_id"]`
But, if during the loop, you come a second time on this line, you will overwrite the value previously set, no ?

Reply:
yep, you're right, it's wrong, my bad